ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/WSQueryEndpoint.java

```diff
@@ -185,178 +130,98 @@ public WSQueryEndpoint(
-    final Principal user = session.getUserPrincipal();
-
-    securityExtension.getAuthorizationProvider().ifPresent(
-        provider -> {
-          try {
-            provider.checkEndpointAccess(user, method, path);
-          } catch (final Throwable t) {
-            log.warn(String.format("User:%s is denied access to Websocket "
-                + "%s endpoint", user.getName(), path), t);
-            throw new KsqlException(t);
-          }
-        }
-    );
-  }
+    final String version = requestParams.get(Versions.KSQL_V1_WS_PARAM);
 
-  private KsqlSecurityContext createSecurityContext(final Principal principal) {
-    final ServiceContext serviceContext;
-
-    if (!securityExtension.getUserContextProvider().isPresent()) {
-      serviceContext = defaultServiceContextFactory.create(ksqlConfig, Optional.empty(),
-          schemaRegistryClientFactory);
-    } else {
-      // Creates a ServiceContext using the user's credentials, so the WS query topics are
-      // accessed with the user permission context (defaults to KSQL service context)
-
-      serviceContext = securityExtension.getUserContextProvider()
-          .map(provider ->
-              serviceContextFactory.create(
-                  ksqlConfig,
-                  Optional.empty(),
-                  provider.getKafkaClientSupplier(principal),
-                  provider.getSchemaRegistryClientFactory(principal)))
-          .get();
-    }
-
-    return new KsqlSecurityContext(Optional.ofNullable(principal), serviceContext);
-  }
-
-  private void validateVersion(final Session session) {
-    final Map<String, List<String>> parameters = session.getRequestParameterMap();
-    activenessRegistrar.updateLastRequestTime();
-
-    final List<String> versionParam = parameters.getOrDefault(
-        Versions.KSQL_V1_WS_PARAM, Collections.singletonList(Versions.KSQL_V1_WS));
-
-    if (versionParam.isEmpty()) {
+    if (version == null) {
       return;
     }
 
-    if (versionParam.size() != 1) {
-      throw new IllegalArgumentException("Received multiple api versions: " + versionParam);
-    }
-
-    if (!versionParam.get(0).equals(Versions.KSQL_V1_WS)) {
-      throw new IllegalArgumentException("Received invalid api version: " + versionParam);
+    if (!Versions.KSQL_V1_WS.equals(version)) {
+      throw new IllegalArgumentException("Received invalid api version: " + version);
     }
   }
 
-  private KsqlRequest parseRequest(final Session session) {
+  private KsqlRequest parseRequest(final MultiMap requestParams) {
     try {
-      final List<String> jsonRequests = session.getRequestParameterMap()
-          .getOrDefault("request", Collections.emptyList());
+      final String jsonRequest = requestParams.get("request");
 
-      if (jsonRequests == null || jsonRequests.isEmpty()) {
-        throw new IllegalArgumentException("missing request parameter");
-      }
-
-      final String jsonRequest = Iterables.getLast(jsonRequests, "");
```

Comment:
Looks like we've changed behavior here since `MultiMap.get()` returns the first element when there are multiple, in contrast to the old behavior here that returned the last element. I don't imagine sending multiple statements in a websocket request is a common use case, but maybe it's worth remaining consistent. Up to you.

Reply:
Behave of multimap.get is to return the last element, so behaviour is the same.
Honestly, I don't people are going to send a query with two requests, and if they do they will soon find out only the last one is being used.

Reply:
Really? The docs say MultiMap.get() returns the first element: https://vertx.io/docs/apidocs/io/vertx/core/MultiMap.html#get-java.lang.String-